rdt/transformers/base.py

```diff
@@ -124,6 +126,9 @@ def _get_columns_data(data, columns):
 
     @staticmethod
     def _set_columns_data(data, columns_data, columns):
+        if isinstance(columns_data, pd.Series) or isinstance(columns_data, pd.DataFrame):
```

Comment:
Using `assign` didn't fix the problem. I also realized that `columns_data` can be a Series, DataFrame or numpy array. So it seemed easiest to just fix the index if `columns_data` has one.

Reply:
@csala I timed both using `to_numpy` and this method, and this seemed to be slightly faster for all `DataFrame` sizes